### PR TITLE
Add frontend draft

### PR DESCRIPTION
**Pull request by cmiacz**

### CODE REVIEW

**Review by KozielPiotr — APPROVED** [review]
I think it should be ok for now.

File: frontend/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "editor.formatOnSave": true
```
Comment (Noapte) [inline frontend/.vscode/settings.json:2]: this file should be part of 'gitignore'

File: frontend/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (Noapte) [inline frontend/public/index.html]: change page title

File: frontend/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <!--
+      This HTML file is a template.
```
Comment (Noapte) [inline frontend/public/index.html]: remove unnecessary comments

File: frontend/src/components/App.test.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import App from "./App";
+
+test("renders learn react link", () => {
```
Comment (Noapte) [inline frontend/src/components/App.test.tsx]: Do you need this test file ?

File: frontend/package.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "react-scripts": "3.4.0",
+    "rxjs": "^6.5.4",
+    "typescript": "^3.7.5",
+    "url-join": "^4.0.1"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
```
Comment (Noapte) [inline frontend/package.json]: Could you add check style (tslint / eslint ... ) to package.json ?

File: frontend/.vscode/settings.json (hunk 1 above)
Comment (KozielPiotr) [reply frontend/.vscode/settings.json:2]: Added

File: frontend/public/index.html (hunk 2 above)
Comment (KozielPiotr) [reply frontend/public/index.html]: Changed to "Shared bills".

File: frontend/public/index.html (hunk 3 above)
Comment (KozielPiotr) [reply frontend/public/index.html]: Removed

File: frontend/src/components/App.test.tsx (hunk 4 above)
Comment (KozielPiotr) [reply frontend/src/components/App.test.tsx]: I don't think so. File removed.

File: frontend/package.json (hunk 5 above)
Comment (KozielPiotr) [reply frontend/package.json]: Done. Could you edit config file for eslint to work better with TS?

Comment (Noapte) [reply frontend/package.json]: for now those default rules should be ok.

**Review by Noapte — APPROVED** [review]

File: frontend/src/services/hooks.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import React from "react";
+
+import { Observable } from "rxjs";
+
+/**
+ * hooks that are used repeatedly
+ */
+class HooksService {
```
Comment (cmiacz) [inline frontend/src/services/hooks.tsx]: Do not put custom hooks into service. There's no point putting it into a class as you use each functions separately. Create separate module for hooks e.g. `src/hooks` and rename this file to `observable.tsx` and then just export `useObservable` function

File: frontend/src/components/App.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import MainPage from "./main/Main";
+import LoginPage from "./login/Login";
+
+/**
+ * Main application function
+ */
+function App() {
+  const [isAuthenticated, setAuthenticated] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
```
Comment (cmiacz) [inline frontend/src/components/App.tsx]: Use `useObservable` hook

File: frontend/src/services/hooks.tsx (hunk 6 above)
Comment (KozielPiotr) [reply frontend/src/services/hooks.tsx]: Done

File: frontend/src/components/App.tsx (hunk 7 above)
Comment (KozielPiotr) [reply frontend/src/components/App.tsx]: Done

**Pull request merged**